test/integration/grpc_management_plane_api_test.go

```diff
@@ -177,16 +187,102 @@ func TestGrpc_ConfigUpload(t *testing.T) {
 	require.NoError(t, err)
 	assert.Equal(t, http.StatusOK, resp.StatusCode())
 
-	client = resty.New()
+	responses := getManagementPlaneResponses(t, 2)
+
+	assert.Equal(t, mpi.CommandResponse_COMMAND_STATUS_OK, responses[0].GetCommandResponse().GetStatus())
+	assert.Equal(t, "Successfully updated all files", responses[0].GetCommandResponse().GetMessage())
+	assert.Equal(t, mpi.CommandResponse_COMMAND_STATUS_OK, responses[1].GetCommandResponse().GetStatus())
+	assert.Equal(t, "Successfully updated all files", responses[1].GetCommandResponse().GetMessage())
+}
+
+func TestGrpc_ConfigApply(t *testing.T) {
+	ctx := context.Background()
+	teardownTest := setupConnectionTest(t, false)
+	defer teardownTest(t)
+
+	nginxInstanceID := verifyConnection(t)
+
+	responses := getManagementPlaneResponses(t, 1)
+	assert.Equal(t, mpi.CommandResponse_COMMAND_STATUS_OK, responses[0].GetCommandResponse().GetStatus())
+	assert.Equal(t, "Successfully updated all files", responses[0].GetCommandResponse().GetMessage())
+
+	// No config changes
+
+	performConfigApply(t, nginxInstanceID)
+
+	// Valid config
+
+	err := mockManagementPlaneGrpcContainer.CopyFileToContainer(
+		ctx,
+		"../config/nginx/nginx-with-test-location.conf",
+		fmt.Sprintf("/mock-management-plane-grpc/config/%s/etc/nginx/nginx.conf", nginxInstanceID),
+		0o666,
+	)
+	require.NoError(t, err)
+
+	performConfigApply(t, nginxInstanceID)
+
+	// Invalid config
+
+	err = mockManagementPlaneGrpcContainer.CopyFileToContainer(
+		ctx,
+		"../config/nginx/invalid-nginx.conf",
+		fmt.Sprintf("/mock-management-plane-grpc/config/%s/etc/nginx/nginx.conf", nginxInstanceID),
+		0o666,
+	)
+	require.NoError(t, err)
+
+	performConfigApply(t, nginxInstanceID)
+
+	responses = getManagementPlaneResponses(t, 5)
+	t.Logf("Config apply responses: %v", responses)
+	assert.Equal(t, mpi.CommandResponse_COMMAND_STATUS_OK, responses[0].GetCommandResponse().GetStatus())
```

Comment:
Should we add a test to check if config apply fails but rollback isn't needed ?

Reply:
let be see if I can add a test for that

Reply:
Happens if file isn't in allowed directories might be the easiest way to try test it

Reply:
Added a new test for that